### PR TITLE
Add iterator to FlagValue

### PR DESCRIPTION
**Pull request by polybassa**
This PR adds the iterator functionality to the FlagValue class. This might be useful. At least I want to use it for some OBD related applications.

### CODE REVIEW

**Comment by polybassa** [comment]
I have such an use case in mind:
```python
resp = sock.sr1(OBD()/OBD_S01(pid=0))
for pid in resp.data_records[0].supportedPIDs:
    req = eval('OBD()/OBD_S01()/OBD_%s()' % pid)
    sock.sr1(req)
    ...
```
This would allow me to easy generate further request packets based on the `FlagsField supportedPIDs`.

File: scapy/fields.py [hunk 1]
```diff
@@ -1814,6 +1814,22 @@ def __str__(self):
             x >>= 1
         return ("+" if self.multi else "").join(r)
 
+    def __iter__(self):
+        self.iter = 0
+        return self
```
[the remark this answers is not in the document]
Comment (polybassa) [reply scapy/fields.py]: Could you please explain the implementation of the `FlagValueIter` object more in detail. I would like to implement it to full-fill your remarks.

File: test/fields.uts [hunk 2]
```diff
@@ -1283,3 +1283,120 @@ a = DebugPacket(addr="scapy.net", atyp=0x3)
...
+class FlagsTest(Packet):
+    fields_desc = [FlagsField("flags", 0, 8,
+                              ["f0", "f1", "f2", "f3", "f4", "f5", "f6", "f7"])]
+
+= Test upper nibble
+
+a = FlagsTest(b"\xf0")
+flags = list()
+for f in a.flags:
+    flags.append(f)
```
[the remark this answers is not in the document]
Comment (polybassa) [reply test/fields.uts]: I will change that. Thanks.

**Comment by polybassa** [comment]
@p-l- Is this PR fine, now?

**Comment by p-l-** [comment]
@polybassa yep! Thank you!

**Pull request merged**